rpc2/example/client.go

```diff
@@ -14,32 +14,33 @@ type GenericClient interface {
 //---------------------------------------------------------------------
 
 type ArithClient struct {
-	cli GenericClient
+	GenericClient
 }
 
 func (a ArithClient) Add(arg AddArgs) (ret int, err error) {
-	err = a.cli.Call("test.1.arith.add", arg, &ret)
+	err = a.Call("test.1.arith.add", arg, &ret)
 	return
 }
 
 func (a ArithClient) Broken() (err error) {
-	err = a.cli.Call("test.1.arith.broken", nil, nil)
+	err = a.Call("test.1.arith.broken", nil, nil)
 	return
 }
 
 //---------------------------------------------------------------------
 
 type Client struct {
+	port int
 }
 
 func (s *Client) Run() (err error) {
 	var c net.Conn
-	if c, err = net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", port)); err != nil {
+	if c, err = net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", s.port)); err != nil {
```

Comment:
I'd prefer not to use package variables across file boundaries: it makes understanding the code and factoring out functionality more difficult.